pallets/liquidity/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2022 Polkadex oü.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{dispatch::DispatchResult, traits::Currency};
+use pallet_timestamp::{self as timestamp};
+use sp_std::prelude::*;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod mock;
+
+//constant proxy value
+const PROXY_ACCOUNT_BYTES: [u8; 32] = [0; 32];
```

Comment:
This can cause issues as the proxy constant address might have already been taken by some other account in the mainnet. The governance should provide bytes of proxy addresses during registration.

Reply:
Yes, derive using PalletId

Reply:
Instead of deriving a proxy using pallet id, we can do the following things. 
- hardcore the proxy and pallet main account. 
- on genesis insert both accounts (proxy and main) directly in the ocex pallet account storage.
- the governance therefore will not need to call register_account as well, the only functionality accessible to governance will be deposit and withdraw.

Reply:
We cannot do that, and we need to support multiple accounts.

Reply:
wana guess what's the private key for public key [0u8; 32]?